program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,88 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+import logging
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from program_intent_engagement.apps.api.serializers import ProgramIntentSerializer
+from program_intent_engagement.apps.core.models import ProgramIntent, User
+
+log = logging.getLogger(__name__)
+
+
+class ProgramIntentsView(APIView):
+    # todo: add expected program intent data
+    """
+    View to insert program intents.
+
+    Given a program intent, this view will either insert the program intent. If the given program intent already
+    exists, no action is taken (no duplicate row).
+
+    Path: /api/[version]/program_intents
+
+    Accepts: [POST]
+
+    HTTP POST
+    Inserts a Program Intent.
+    Expected POST data:
+    POST data parameters:
+     * program_uuid: The uuid of the program intent.
+     * reason: The reason for how intent was measured.
+     * certainty: The certainty of the intent for the user.
+     * effective_timestamp: When the event occurred that created the intent.
+    Returns:
+     * 200: The program intent was inserted.
+     * 400: The program intent request was invalid.
+     * 409: Duplicate program intent was not inserted.
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    # todo: do I need to add these? look into what these do
+    # serializer_class = ProgramIntentSerializer
+    # queryset = ProgramIntent.objects.all
+    # lookup_field = "user__lms_user_id"
+
+    @classmethod
+    def add_program_intent_user(cls, request):
+        # get internal user from the LMS user id in the request
+        user = User.objects.get(lms_user_id=request.user.lms_user_id)
+
+        # swap LMS user id for the internal user id before passing to serializer
+        request.data["user_id"] = user.id
+
+        return ProgramIntentSerializer(data=request.data)
+
+    def post(self, request):
+        """
+        Create and insert a program intent. Duplicates not inserted.
+        """
+        request_data = request.data
+
+        if request.user is None:
```

Comment:
this happens when it fails to decode the jwt token and if we include the permission class I think we can never get here

Reply:
Ah, because we're extracting the `lms_user_id` and the permission check would happen before we make that change to user_id, right, @ashultz0 ?

Reply:
Oh wait, nvm! I think you're saying this is extraneous/not needed since jwt should handle?

Reply:
well we expect the permission check to have freaked out way before this if request.user was not set up out of the jwt, so this check is extraneous

if we expected user to come in none sometimes it's worth having a check, but if it can only happen if all our other expectations are violated it just makes the code more confusing to handle it

Reply:
Removed this since this is not needed. All resolved, thanks @ashultz0 for catching this!